### PR TITLE
Created migration for loans, also modified Id to ID in loan.go

### PR DESCRIPTION
**Pull request by bokimilinkovic**
https://trello.com/c/nAcxrW6f/115-test-errorhandler

### CODE REVIEW

File: domain/model/loan.go [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
 }
 
-func NewLoan(userId int, bookId int, state loanType) *Loan{
-	uuid ,err := uuid.NewUUID()
-	if err != nil{
-		fmt.Println(err)
+func NewLoan(userId int, bookId int, state loanType) *Loan {
+	uuid, err := uuid.NewUUID()
+	if err != nil {
+		return nil
```
Comment (sfqi) [inline domain/model/loan.go]: got to return the error as well

File: migrations/4_create_loan_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE loans(
+    id serial primary key,
+    transaction_id varchar(255) unique,
```
Comment (sfqi) [inline migrations/4_create_loan_table.up.sql]: not unique, every borrowed-returned pair will have the same transaction id

File: migrations/4_create_loan_table.up.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    user_id int not null,
+    book_id int unique not null,
```
Comment (sfqi) [inline migrations/4_create_loan_table.up.sql]: not unique, the same book can be loaned multiple times

File: domain/model/loan.go [hunk 4]
```diff
@@ -1,44 +1,44 @@
...
 		TransactionID: uuid.String(),
-		UserID:userId,
-		BookID:bookId,
-		Type:state,
-		CreatedAt : time.Now(),
-	}
+		UserID:        userId,
+		BookID:        bookId,
+		Type:          state,
+		CreatedAt:     time.Now(),
```
Comment (sfqi) [inline domain/model/loan.go]: I think gorm sets the value of `CreatedAt`. If so, we don't have to do it here. Can you double-check?

Comment (bokimilinkovic) [reply domain/model/loan.go]: yes, you are right, it sets automatically

Comment (sfqi) [reply domain/model/loan.go]: probably fine if you remove it, then

**Pull request merged**